exercises/crypto-square/src/example.c

```diff
@@ -0,0 +1,92 @@
+#include "crypto_square.h"
+#include <stdlib.h>
+#include <ctype.h>
+#include <stdio.h>
+#include <math.h>
+
+/* mstrlen: get the length of input counting only alnums */
+static int mstrlen(const char *input)
+{
+   int i = 0;
+   while (*input) {
+      if (isalnum(*input))
+         i++;
+      input++;
+   }
+
+   return i;
+}
+
+/* get_cols_and_rows: calculate columns and rows from len */
+static void get_cols_and_rows(int len, int *cols, int *rows)
+{
+   *cols = sqrt(len);
+
+   if (*cols * *cols == len) {
+      *rows = *cols;
+   } else if (*cols * (*cols + 1) >= len) {
+      *rows = *cols;
+      (*cols)++;
+   } else {
+      (*cols)++;
+      *rows = *cols;
+      if ((*cols * *cols) < len)
+         (*cols)++;
+   }
+}
+
+/* get_pos_cipher: calculate the position in the cipher string */
+static int get_pos_cipher(const int pos_inp, const int rows, const int cols)
+{
+   return (pos_inp / cols) + (pos_inp % cols) * (rows + 1);
+}
+
+char *ciphertext(const char *input)
+{
+   if (input == NULL)
+      return NULL;
+   if (*input == '\0')
+      return calloc(1, 1);
+
+   int len_inp = mstrlen(input);
+   if (len_inp == 0)
+      return calloc(1, 1);
+
+   int cols, rows;
+   get_cols_and_rows(len_inp, &cols, &rows);
+
+   /* no need for extra row if single line */
+   int len_out = rows > 1 ? (rows + 1) * cols : rows * cols;
+
+   char *res = malloc(len_out + 1);
```

Comment:
You may want to be explicit about the allocation size here, instead writing the malloc as `malloc(sizeof(char) * (len_out + 1))`. You're using it correctly, but misusing malloc or calloc calls by mixing up what the arguments should be is a very common problem.

Reply:
I can not really follow here - was not ```sizeof(char)``` defined in the standard to be 1? Or you mean to write it explicit for the sake of readability?